src/declarations/stencil-public-compiler.ts

```diff
@@ -946,18 +946,30 @@ export interface SerializeDocumentOptions extends HydrateDocumentOptions {
    */
   removeHtmlComments?: boolean;
   /**
-   * If set to `true` the component will be rendered within a Declarative Shadow DOM.
-   * If set to `false` Stencil will ignore the contents of the shadow root and render the
-   * element as given in provided template.
-   * @default true
-   */
-  serializeShadowRoot?: boolean;
+   * - If set to `dsd` the component will be rendered within a Declarative Shadow DOM.
+   * - If set to `scoped` Stencil will render the contents of the shadow root as a `scoped: true` component
+   * and the shadowDOM will be created during client-side hydration.
+   * - Alternatively you can mix and match the two by providing an object with `dsd` and `scoped` keys, 
+   * the value arrays containing tag names of the components that should be rendered in that mode.
+   *
+   * *NOTE* `true | false` values have been deprecated in favor of `dsd` and `scoped`
+   * @default 'dsd'
+   */
+  serializeShadowRoot?:
+    | 'dsd'
+    | 'scoped'
+    | {
+        dsd: string[];
+        scoped: string[];
+        default: 'dsd' | 'scoped';
+      }
```

Comment:
Let's add 2 examples to the JSDoc, one defining the value as e.g. `scoped` , the other showing an example of having 2 components as DSD and e.g. 1 component as scoped